main.tf

```diff
@@ -0,0 +1,132 @@
+terraform {
+
+
+  required_providers {
+    aws = {
+      source = "hashicorp/aws"
+    }
+  }
+
+ backend "s3" {
+   bucket = "custom-terraform-state"
+   key = "dev/s3/terraform.tfstate"
+   region= "us-west-2"
+   encrypt = true
+ }
+ 
+
+}
+
+provider "aws" {
+  region = var.selected_region
+}
+
+
+resource "aws_key_pair" "key-module" {
+  key_name   = "key-module"
+  public_key = var.pub_key
+}
+/*
+resource "aws_s3_bucket" "terraform_state" {
+  bucket = "custom-terraform-state"
+  lifecycle {
+  prevent_destroy = true
+  }
+
+  versioning {
+  enabled = true
+  }
+
+  server_side_encryption_configuration {
+  rule {
+  apply_server_side_encryption_by_default {
+  sse_algorithm = "AES256"
+  } 
+  }
+  }
+}
+*/
```

Comment:
```suggestion
```
I'm not familiar with terraform at all, but would it be possible to remove these lines?